StaticInstaller.php

```diff
@@ -123,7 +159,86 @@ public static function composerFinishUpdate(Event $composer_event = null)
         
         return $text ? $text : 'No apps to update' . ".\n";
     }
+    /**
+     * Unlink backup from specified backup folder, folder name is defined by backupTime-String
+     *
+     * @param string $app_alias
+     * @param string $backupTime
+     * @return boolean return TRUE if unlinking BackUp was successful, return FALSE if it was not
+     */
+    public function unlinkBackup($app_alias, $backupTime){
+        $exface = $this->getWorkbench();
+        $app = $exface->getApp(self::PACKAGE_MANAGER_APP_ALIAS);
+        try {
+            $link = $app->getWorkbench()->filemanager()->getPathToBaseFolder().DIRECTORY_SEPARATOR.Filemanager::FOLDER_NAME_BACKUP.DIRECTORY_SEPARATOR.$backupTime.DIRECTORY_SEPARATOR.str_replace(".",DIRECTORY_SEPARATOR,$app_alias);
+            Filemanager::deleteDir($link);
+            // Delete Parent folders to avoid clutter, provided that they are in fact empty
+            $parentLink = explode(".",$app_alias);
+            $parentLink = $app->getWorkbench()->filemanager()->getPathToBaseFolder().DIRECTORY_SEPARATOR.Filemanager::FOLDER_NAME_BACKUP.DIRECTORY_SEPARATOR.$backupTime.DIRECTORY_SEPARATOR.$parentLink[0].DIRECTORY_SEPARATOR;
+            if (Filemanager::isDirEmpty($parentLink)){
+                Filemanager::deleteDir($parentLink);
+            }
+            self::printToStdout("Delete unused ".$app_alias." app backup\n");
+        }
+        catch(\Exception $e){
+            self::printToStdout("Could not delete ".$app_alias.". ".$e->getMessage() . ' in ' . $e->getFile() . ' at line ' . $e->getLine() . ".\n");
+            return false;
+        }
+        return true;
+    }
+    /**
+     * Backup all apps that are listen in composer.json but not listed in EXCLUDE_BACKUP_PACKAGES
+     * since they have no own backup function to call to
+     *
+     * @param Event $composer_event
+     * @return Event $composer_event
+     */
+    public static function composerBackupEverything(Event $composer_event = null){
+        $composerContent = file_get_contents('composer.json');
+        $obj = json_decode($composerContent);
+        $installer = new self();
 
+        //write consistent backuptime to delete excess data after update run
+        $backupTime = date('Y_m_d_H_i');
+        $temp = self::getTempFile();
+        $temp['backupTime'] = $backupTime;
+        self::setTempFile($temp);
+        foreach($obj->require as $requiredPackage => $packageInfo){
+
+            $appIdentifier = explode("/",$requiredPackage);
```

Comment:
Nicht jedes Composer-Package ist eine App. Das erkennst du an den folgendem Abschnitt in der composer.json 
```
    "extra": {
        "app": {
            "app_uid": "0x31000000000000000000000000000000",
            "app_alias": "exface.Core",
            "model_md5": "c4d5265d88a0eb9c566e82120d523058",
            "model_timestamp": "2017-11-07 18:01:52"
        }
    }
```
Hier bekommst du auch den korrekten Alias der App her. Der Alias muss nicht mit dem Paket-Namen übereinstimmen. Der Alias ist eine (kurze) Sonderform des Namespace - und dieser kann in composer-Paketen beliebig sein und wird in composer.json unter autoload definiert.